tensorflow/python/tools/saved_model_cli.py

```diff
@@ -137,24 +146,119 @@ def _show_inputs_outputs(saved_model_dir, tag_set, signature_def_key, indent=0):
   for input_key, input_tensor in sorted(inputs_tensor_info.items()):
     in_print('  inputs[\'%s\'] tensor_info:' % input_key)
-    _print_tensor_info(input_tensor, indent+1)
+    _print_tensor_info(input_tensor, indent + 1)
 
   in_print('The given SavedModel SignatureDef contains the following '
            'output(s):')
   for output_key, output_tensor in sorted(outputs_tensor_info.items()):
     in_print('  outputs[\'%s\'] tensor_info:' % output_key)
-    _print_tensor_info(output_tensor, indent+1)
+    _print_tensor_info(output_tensor, indent + 1)
 
   in_print('Method name is: %s' %
            meta_graph_def.signature_def[signature_def_key].method_name)
 
 
+def _show_defined_functions(saved_model_dir, indent=0):
+  """Prints the function definition of SavedModel2.0 located at saved_model_dir
+
+     Args:
+       saved_model_dir: Directory containing the SavedModel to inspect.
+       indent: How far (in increments of 2 spaces) to indent each line of output.
+  """
+  with ops_lib.Graph().as_default():
+    trackable_object = load.load(saved_model_dir)
+  indent_str = '  ' * indent
+
+  def in_print(s):
+    print(indent_str + s)
+
+  print('Defined Functions:')
+  functions = save._AugmentedGraphView(
+      trackable_object).list_functions(trackable_object)
+  for name, function in functions.items():
+    in_print('Function Name: \'%s\'' % name)
+    for index, concrete_functions in enumerate(
+            function._list_all_concrete_functions_for_serialization(), 1):
+      args, kwargs = (concrete_functions.structured_input_signature)
+      in_print('Option #%d' % index)
+      in_print('  Callable with:')
+      _print_args(args, indent=3)
+      if kwargs:
+        _print_args(kwargs, "Named Argument", indent=3)
+
+
+def _print_args(arguments, argument_type="Argument", indent=0):
+  """Formats and prints the argument of the concrete functions defined in the model
+
+     Args:
+       arguments: Arguments of the concrete functions.
+       argument_type: Type of Argument List to Format and print.
+       indent: How far (in increments of 2 spaces) to indent each line of output.
+  """
+  indent_str = '  ' * indent
+
+  def quotes(value):
+    is_quotes = '\'' * isinstance(value, str)
+    return is_quotes + value + is_quotes
+
+  def in_print(s, end='\n'):
+    print(indent_str + s, end=end)
+
+  def is_nested(args):
+    return nest.is_nested(args) and not isinstance(args, dict)
+  if is_nested(arguments):
+    for index, element in enumerate(arguments, 1):
+      if indent == 3:
+        in_print('%s #%d' % (argument_type, index))
+      if isinstance(element, tensor_spec.TensorSpec):
+        _print_tensor_spec(element, indent)
+      elif is_nested(element):
+        in_print('  DType: %s' % type(element).__name__)
+        in_print('  Values: [', end='')
+        _print_args(element, indent + 1)
+        in_print('  ]')
+      elif isinstance(element, dict):
```

Comment:
> Why don't we print dictionaries? What if the function returns a dictionary?

@vbardiovskyg , we are printing dictionaries, that is needed since, dictionary is nested and it is the only simple structure that has key-value pair, and one of the values might be list or with much deeper nesting, so I'm handling it differently.